### ISSUE TITLE
C++14 support

### ISSUE THREAD

**Issue by dakeryas** [project8/scarab#7]
My ROOT6 library was built with the C++14 standard. 

As a result, I cannot build Katydid without using set( CMAKE_CXX_STANDARD 14 ) in Nymph/Scarab/cmake/PackageBuilder.cmake (and in Source/Time/Monarch/Scarab/cmake/PackageBuilder.cmake). Indeed, with the C++11 standard (Katydid default) I obtain the following error from GCC 6.2.0

> [ 35%] Building CXX object Source/Utility/CMakeFiles/KatydidUtility.dir/__/__/Examples/KTProcessorTemplate.cc.o
> In file included from /usr/include/c++/6/experimental/string_view:39:0,
>                 from /home/vsibille/Software/root/include/RStringView.h:24,
>                 from /home/vsibille/Software/root/include/TString.h:28,
>                 from /home/vsibille/Software/root/include/TNamed.h:26,
>                 from /home/vsibille/Software/root/include/TGraph.h:24,
>                 from /home/vsibille/Software/root/include/TSpline.h:15,
>                 from /home/vsibille/Documents/project8/katydid/Source/Utility/KTSpline.hh:12,
>                 from /home/vsibille/Documents/project8/katydid/Source/Utility/KTSpline.cc:8:
> /usr/include/c++/6/bits/c++14_warning.h:32:2: error: #error This file requires compiler and library support for the forthcoming ISO C++ 2014 standard. This support is currently experimental, and must be enabled with the -std=c++1y or -std=gnu++1y compiler options.
> #error This file requires compiler and library support for the forthcoming \

I reckon it would be useful to have a C++14 option in the CMake files, defaulting to OFF.

A snippet is proposed below
```
# flag for using C++14
option( USE_CPP14 "Flag for building with C++14" OFF )
if( USE_CPP14 )
  add_definitions( -DUSE_CPP14 )
  set( CMAKE_CXX_STANDARD 14 )
else( USE_CPP14 )
  remove_definitions( -DUSE_CP14 )
  set( CMAKE_CXX_STANDARD 11 )
endif( USE_CPP14 )
```

**Comment by nsoblath** [comment]
I've pushed a Katydid branch called feature/cpp14 that includes the changes in Scarab to build with C++14.  It builds on my machine.  Can you please test it and let me know if it works for you?

After checking out the branch, be sure to run `git submodule update --recursive` to get the submodule (Nymph and Scarab) updates.

**Comment by dakeryas** [comment]
Thank you! 

The branch origin/feature/cpp14 builds smoothly!

**Comment by nsoblath** [comment]
Now available in develop at commit fd5bf19990f6ff979b03984b59a9c0a3f9800a1d
